### PR TITLE
Complete the Hibernate ORM guide, clarify the limitations

### PR DESCRIPTION
**Pull request by Sanne**
long overdue, finally clarifying the limitations we still have.

### CODE REVIEW

File: docs/src/main/asciidoc/hibernate-orm.adoc [hunk 1]
```diff
@@ -385,6 +431,85 @@ Finally, the second-level cache can be disabled globally by setting `hibernate.c
...
+
+Transaction Manager integration::
+You don't need to set this up, Quarkus automatically injects the reference to the Narayana Transaction Manager.
+The dependency is included automatically as a transitive dependency of the Hibernate ORM extension.
+All configuration is optional; for more details see link:transaction.html[Using Transactions in Quarkus].
+
+Connection pool::
+Don't need to choose one either. Quarkus automatically includes the Agroal connection pool;
+just configure your datasource as in the above examples and it will setup Hibernate ORM to use Agroal.
+More details about this connection pool can be found in link:datasource-guide.html[Quarkus - Datasources].
```
Comment (gsmet) [inline docs/src/main/asciidoc/hibernate-orm.adoc]: This URL is outdated, it should point to: https://quarkus.io/guides/datasource

There is a redirect so it's no biggie but better get this straight.

**Review by gsmet — CHANGES_REQUESTED** [review]
Very nice work, thanks a lot for that.

There's just one URL to adapt and it's good to go.

Thanks!

File: docs/src/main/asciidoc/hibernate-orm.adoc (hunk 1 above)
Comment (Sanne) [reply docs/src/main/asciidoc/hibernate-orm.adoc]: ah thanks!  well, without the `.html` too :)

Comment (gsmet) [reply docs/src/main/asciidoc/hibernate-orm.adoc]: Yeah, I will make a pass on all the guides to remove the `.html`. That's why I didn't mention it.

**Comment by Sanne** [comment]
@gsmet applied rebased, push forced.

I stripped all .html also from a couple other cross-guide links this had.

**Review by gsmet — APPROVED** [review]

**Pull request merged**